getconf/base.py

```diff
@@ -23,7 +23,46 @@
     logger.addHandler(compat.NullHandler())
 
 
-ConfigKey = collections.namedtuple('ConfigKey', ['section', 'entry', 'envvar', 'doc'])
+_ConfigKey = collections.namedtuple(
+    'ConfigKey',
+    ['section', 'entry', 'envvar', 'doc', 'default', 'type_hint']
+)
+
+
+class ConfigKey(_ConfigKey):
+
+    def __hash__(self):
+        if self.type_hint == 'list' and self.default is not None:
+            # Make sure we can hash default
+            default = tuple(self.default)
+        else:
+            default = self.default
+        return hash((self.section, self.entry, self.envvar, self.doc, default, self.type_hint))
+
+    def as_ini_entry(self):
+        """Format as a commented entry in INI format (section is omitted)
+
+        ; type=str - The provided documentation
+        ;entry = default_value
+        """
+        # Remove any new line in self.doc
+        one_line_doc = ' '.join(self.doc.split())
+        doc_part = ' - %s' % one_line_doc if one_line_doc else ''
```

Comment:
Does this parse as ``(' - %s' % one_line_doc) if one_line_doc else ''`` or ``' - %s' % (one_line_doc if one_line_doc else '')``?